src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -150,18 +228,19 @@ public void StartRecording()
                 // find the first index number we haven't used yet
                 do
                 {
-                    _currentVideoDirectory =
+                    _currentGameplaySessionDirectoryPrefix =
                         $"{stateRecordingsDirectory}/{Application.productName}/run_{_videoNumber++}";
-                } while (Directory.Exists(_currentVideoDirectory) || File.Exists(_currentVideoDirectory + ".zip"));
+                } while (Directory.Exists(_currentGameplaySessionDirectoryPrefix + "_data") || File.Exists(_currentGameplaySessionDirectoryPrefix + "_data.zip"));
 
-                if (!Directory.Exists(_currentVideoDirectory))
+                if (!Directory.Exists(_currentGameplaySessionDirectoryPrefix))
                 {
-                    Directory.CreateDirectory(_currentVideoDirectory);
+                    Directory.CreateDirectory(_currentGameplaySessionDirectoryPrefix + "_data");
```

Comment:
This isn't properly checking that the inner folders exist

Reply:
This will create all directories needed up to that folder, so it handles the parents as well.
> Creates all directories and subdirectories in the specified path unless they already exist.

Reply:
yes.. BUT.. checking that the parent folder exists does not confirm that the inner folders also exist, and we don't call to create the inner if the parent exists

Reply:
Could `RGEditorUtils.CreateAllAssetFolders(targetDirPath);` be the solution here?

Reply:
unfortunately not.. this needs to work at runtime and those utils utilize mostly assetdatabase apis (editor only)

Reply:
I see what you are saying now... I realize why overcomplicate it, and just call create directory no matter what since it will create if it doesn't exist, and be a no-op otherwise... Fixed!